storage/remote/codec_test.go

```diff
@@ -145,3 +147,59 @@ func TestConcreteSeriesClonesLabels(t *testing.T) {
 	gotLabels = cs.Labels()
 	require.Equal(t, lbls, gotLabels)
 }
+
+func TestDuplicateLabelsSimple(t *testing.T) {
```

Comment:
you could turn this and `TestDuplicateLabelsDoNothing` into a single test by using a subtest and turning the input/expected output into a struct

Reply:
I removed these tests and replaced them with more test cases in `TestValidateLabelsAndMetricName`. I'll see if I can refactor that test to use subtests though.

Reply:
I refactored `TestValidateLabelsAndMetricName` to use subtests, let me know if it looks right to you.